contracts/proofs/ERC721Validator.sol

```diff
@@ -0,0 +1,182 @@
+  bytes32 constant public APPROVAL_EVENT_SIGNATURE = 0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925;
+  // keccak256('Deposit(address,address,uint256,uint256,uint256)')
+  bytes32 constant public DEPOSIT_EVENT_SIGNATURE = 0x4e2ca0515ed1aef1395f66b5303bb5d6f1bf9d61a353fa53f73f8ac9973fa9f6;
+  // keccak256('Withdraw(address,address,uint256,uint256,uint256)')
+  bytes32 constant private WITHDRAW_EVENT_SIGNATURE = 0xebff2602b3f468259e1e99f613fed6691f3a6526effe6ef3e768ba7ae7a36c4f;
+  // keccak256('LogTransfer(address,address,address,uint256,uint256,uint256,uint256)') ERC721 log transfer
+  bytes32 public LOG_TRANSFER_EVENT_SIGNATURE = 0x89a7442af1c60542680034769f1d7362e2bf7f1dcff121c3a04df897c1735d92;
+
+  // validate ERC20 TX
+  function validateTransferTx(
+    bytes transferTx
+  ) public {
+    // validate transfer tx
+    RLP.RLPItem[] memory txData = transferTx.toRLPItem().toList();
+
+    // validate ERC20 transfer tx
+    if (!_validateTransferTx(txData)) {
+      IRootChain(rootChain).slash();
+    }
+  }
+
+  function _validateTransferTx(
+    RLP.RLPItem[] memory txData
+  ) internal returns (bool) {
+    // validate tx receipt existence
+    require(
+      validateTxReceiptExistence(
+        txData[0].toUint(), // headerNumber
+        txData[1].toData(), // headerProof,
+
+        txData[2].toUint(), // blockNumber,
+        txData[3].toUint(), // blockTime,
+
+        txData[4].toBytes32(), // txRoot,
+        txData[5].toBytes32(), // receiptRoot,
+        txData[6].toData(), // path,
+
+        txData[7].toData(), // txBytes,
+        txData[8].toData(), // txProof
+
+        txData[9].toData(), // receiptBytes,
+        txData[10].toData() // receiptProof
+      )
+    );
+    return _validateTransferTx(txData[7].toData(), txData[9].toData());
+  }
+  
+  function _validateTransferTx(
+    bytes txData,
+    bytes receiptData
+  ) internal returns (bool) {
+    // check transaction
+    RLP.RLPItem[] memory items = txData.toRLPItem().toList();
+    require(items.length == 9);
+
+    // check if child token is mapped with root tokens
+    address childToken = items[3].toAddress();
+    require(depositManager.reverseTokens(childToken) != address(0));
+
+    // check if transaction is transfer tx
+    // <4 bytes transfer event,address (32 bytes),address (32 bytes), tokenId (32 bytes)>
+    bytes memory dataField = items[5].toData();
+    require(BytesLib.toBytes4(BytesLib.slice(dataField, 0, 4)) == TRANSFER_SIGNATURE);
+
+    // if data field is not 100(4+32+32+32) bytes, return
+    if (dataField.length != 100) {
+      return false;
+    }
+    // sender
+    address sender = getTxSender(items);
+
+    /*
+      check receipt and data field
+      Receipt -->
+        [0]
+        [1]
+        [2]
+        [3]-> [
+          [child token address, [TRANSFER_EVENT_SIGNATURE, from, to, tokenId], <>],
+          [child token address, [LOG_TRANSFER_EVENT_SIGNATURE,token,from,to], <tokenId,input1,input2,tokenId>]
```

Comment:
Remove multiple `tokenId` from receipt.

Reply:
yup, that will be removed once I refactor that withdraw manager code(proofs validation functions).